### PR TITLE
feat: adds a log statement for all URLs for which more than 5 Keywords are selected

### PR DESCRIPTION
**Pull request by chetanbasuray**

### CODE REVIEW

**Review by idoshamun — CHANGES_REQUESTED** [review]
The log statement looks great. But it's better to not create a log instance on every call. So if you can add the logger as a parameter to the function it would be great

**Comment by chetanbasuray** [comment]
> The log statement looks great. But it's better to not create a log instance on every call. So if you can add the logger as a parameter to the function it would be great

Updated

File: src/routes/private.ts [hunk 1]
```diff
@@ -8,14 +8,16 @@ import {
...
 
 export default async function (fastify: FastifyInstance): Promise<void> {
   fastify.post<{ Body: AddPostData }>('/newPost', async (req, res) => {
     if (!req.service) {
       return res.status(404).send();
     }
     const con = getConnection();
-    const operationResult = await addNewPost(con, req.body);
+    const logger = pino();
+    const operationResult = await addNewPost(con, req.body, logger);
```
Comment (idoshamun) [inline src/routes/private.ts:20]: The request object should already have a logger defined. No need to define a new instance.
In most cases you don't have to define an instance as a rule of thumb

**Review by idoshamun — CHANGES_REQUESTED** [review]

File: src/routes/private.ts (hunk 1 above)
Comment (chetanbasuray) [reply src/routes/private.ts]: Updated to use the logger from the req object.

**Review by idoshamun — APPROVED** [review]
Perfect!

**Pull request merged**